controllers/managedocs_controller.go

```diff
@@ -255,3 +323,104 @@ func (r *ManagedOCSReconciler) updateComponents() v1.ComponentStatusMap {
 
 	return components
 }
+
+func (r *ManagedOCSReconciler) isUninstallValid() bool {
+	cmNamespaceName := types.NamespacedName{
+		Name:      r.DeleteConfigMapName,
+		Namespace: r.namespace,
+	}
+
+	configmap := &corev1.ConfigMap{}
+	err := r.Get(r.ctx, cmNamespaceName, configmap)
+	if err == nil {
+		labels := configmap.Labels
+		if _, ok := labels[r.DeleteConfigMapLabelKey]; !ok {
+			return false
+		}
+		return true
+	}
+	return false
+}
+
+func (r *ManagedOCSReconciler) removeOlmComponents() (reconcile.Result, error) {
+
+	r.Log.Info("Deleting subscription")
+	if result, err := r.deleteResource(r.Client, &operators.Subscription{
+		ObjectMeta: metav1.ObjectMeta{Name: r.AddonSubscriptionName, Namespace: r.namespace},
+	}, r.AddonSubscriptionName); err != nil {
```

Comment:
As a matter of readability, let's not create and inline structs inside if conditions.

```suggestion
	subscription :=  &operators.Subscription{
		ObjectMeta: metav1.ObjectMeta{
			Name: r.AddonSubscriptionName, 
			Namespace: r.namespace,
		},
	}
	if result, err := r.deleteResource(r.Client, subscription, r.AddonSubscriptionName); err != nil {
```